### PR TITLE
Improve crate layout, mark deprecations, resolve many test warnings

### PR DESCRIPTION
**Pull request by timvisee**
A lot of refactoring all over the place, some key differences include:

- changed type layout, expose `Qdrant`, `QdrantConfig`, `Error` and other types at crate root
- deprecated old types
- updated `examples/search.rs`, kept old version in `examples/deprecated_search.rs`
- many fixes in doctests

The current documentation can be seen by running `cargo doc --no-deps --open`.

It currently looks like this:

image

### All Submissions:

* [x] Contributions should target the `dev` branch. Did you create your branch from `dev`?
* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### CODE REVIEW

File: examples/search.rs [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
+use qdrant_client::{Qdrant, QdrantConfig, Result};
 
 #[tokio::main]
 async fn main() -> Result<()> {
     // Example of top level client
     // You may also use tonic-generated client from `src/qdrant.rs`
-    let config = QdrantClientConfig::from_url("http://localhost:6334");
-    let client = QdrantClient::new(Some(config))?;
+    let config = QdrantConfig::from_url("http://localhost:6334");
+    let client = Qdrant::new(Some(config))?;
```
Comment (generall) [inline examples/search.rs:14]:
```suggestion
    let client = Qdrant::from_url("http://localhost:6334").build()?;
```

File: examples/search.rs [hunk 2]
```diff
@@ -54,17 +52,17 @@ async fn main() -> Result<()> {
...
-
+    let search_result = client
+        .search_points(
+            SearchPointsBuilder::new(collection_name, [11.; 10], 10)
+                .filter(Filter::all([Condition::matches("bar", 12)]))
+                .with_payload(true)
+                .params(SearchParamsBuilder::default().exact(true)),
+        )
+        .await?;
     dbg!(&search_result);
```
Comment (generall) [inline examples/search.rs:66]: Might be nice to show conversion to json here, will also give is smaller comment

File: examples/search.rs (hunk 1 above)
Comment (timvisee) [reply examples/search.rs]: Tackled in 74ef1441bde64143164d0e5189d6b100f3178144

**Review by generall — APPROVED** [review]

**Review by JojiiOfficial — APPROVED** [review]

File: examples/search.rs (hunk 2 above)
Comment (timvisee) [reply examples/search.rs:64]: Done, now showing it like this:

```rust
    // search_result = [
    //   {
    //     "id": 0,
    //     "version": 0,
    //     "score": 1.0000001,
    //     "payload": {
    //       "bar": 12,
    //       "baz": {
    //         "qux": "quux"
    //       },
    //       "foo": "Bar"
    //     }
    //   }
    // ]
```

**Pull request merged**